File: C#.Net/JDI/Web/JDI Web/Selenium/Elements/Composite/WebSite.cs
using System;
using Epam.JDI.Core.Base;
using JDI_Web.Selenium.Base;
using JDI_Web.Selenium.DriverFactory;
using JDI_Web.Settings;
using OpenQA.Selenium;

namespace JDI_Web.Selenium.Elements.Composite
{
    public class WebSite : Application
    {
        public IWebDriver WebDriver => WebSettings.WebDriverFactory.GetDriver(DriverName);
        public string Url => WebDriver.Url;
        public string BaseUrl => new Uri(WebDriver.Url).GetLeftPart(UriPartial.Authority);
        public string Title => WebDriver.Title;
        private static WebCascadeInit CascadeInit => new WebCascadeInit();

        public static void Init(Type siteType)
        {
            CascadeInit.InitStaticPages(siteType, WebSettings.WebDriverFactory.CurrentDriverName);
            CurrentSite = siteType;
        }

        public static T Init<T>(Type siteType, string driverName) where T : Application
        {
            return CascadeInit.InitPages<T>(siteType, driverName);
        }

        public static T Init<T>(Type siteType, DriverTypes driverType = DriverTypes.Chrome) where T : Application
        {
            return Init<T>(siteType, WebSettings.UseDriver(driverType));
        }

        public T Init<T>(string driverName) where T : Application
        {
            DriverName = driverName;
            return Init<T>(GetType(), driverName);
        }

        public T Init<T>(DriverTypes driverType = DriverTypes.Chrome) where T : Application
        {
            DriverName = WebSettings.UseDriver(driverType);
            return Init<T>(DriverName);
        }

        public T InitScope<T>(object scope) where T : Application
        {
            return Init<T>(GetType());
        }

        public static void Open()
        {
            WebSettings.WebDriver.Navigate().GoToUrl(WebSettings.Domain);
        }

        public void OpenUrl(string url)
        {
            WebDriver.Navigate().GoToUrl(url);
        }

        public void OpenBaseUrl()
        {
            WebDriver.Navigate().GoToUrl(BaseUrl);
        }

        public void Refresh()
        {
            WebDriver.Navigate().Refresh();
        }

        public void Forward()
        {
            WebDriver.Navigate().Forward();
        }

        public void Back()
        {
            WebDriver.Navigate().Back();
        }
    }
}

File: Java/JDI/jdi-uitest-web/src/main/java/com/epam/jdi/uitests/web/selenium/elements/composite/WebSite.java
package com.epam.jdi.uitests.web.selenium.elements.composite;
/*
 * Copyright 2004-2016 EPAM Systems
 *
 * This file is part of JDI project.
 *
 * JDI is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * JDI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; 
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with JDI. If not, see <http://www.gnu.org/licenses/>.
 */


import com.epam.jdi.uitests.core.interfaces.Application;
import com.epam.jdi.uitests.web.selenium.elements.WebCascadeInit;
import com.epam.jdi.uitests.web.selenium.elements.pageobjects.annotations.JSite;

import static com.epam.jdi.uitests.web.selenium.driver.DriverTypes.CHROME;
import static com.epam.jdi.uitests.web.settings.WebSettings.*;
import static org.apache.commons.lang3.StringUtils.isNotBlank;

/**
 * Created by Roman_Iovlev on 8/30/2015.
 */
public class WebSite extends Application {

    public static <T> void init(String driverName, Class<T>... sites) {
        for (Class<T> site : sites) {
            if (site.isAnnotationPresent(JSite.class)) {
                String value = site.getAnnotation(JSite.class).value();
                if (isNotBlank(value)) DOMAIN = value;
            }
            new WebCascadeInit().initStaticPages(site, driverName);
        }
        currentSite = sites[sites.length-1];
    }
    public static <T> void init(Class<T>... sites) {
        if (!getDriverFactory().hasDrivers())
            useDriver(CHROME);
        String driverName = getDriverFactory().currentDriverName();
        init(driverName, sites);
    }

    /**
     * Open page, defined in @JSite, without need to call WebSite.WebPage.open() method
     */
    public static void open(){
        WebPage page = new WebPage(DOMAIN);
        page.setName(currentSite.getSimpleName());
        page.open();
    }
    public static void shouldBeOpened(){
        if (!getDriver().getCurrentUrl().contains(DOMAIN))
            open();
    }
}
